tests/atom.rs

```diff
@@ -45,6 +46,72 @@ fn set_if_none() {
     );
 }
 
+#[test]
+fn compare_and_swap() {
+    cas_test_helper(|a, cas_val, next_val| a.compare_and_swap(cas_val, next_val, Ordering::SeqCst));
+}
+
+#[test]
+fn compare_exchange() {
+    cas_test_helper(|a, cas_val, next_val| {
+        a.compare_exchange(cas_val, next_val, Ordering::SeqCst, Ordering::SeqCst)
+    });
+}
+
+#[test]
+fn compare_exchange_weak() {
+    cas_test_helper(|a, cas_val, next_val| {
+        a.compare_exchange_weak(cas_val, next_val, Ordering::SeqCst, Ordering::SeqCst)
+    });
+}
+
+fn cas_test_helper(
```

Comment:
A few more simple, deterministic tests would be quite nice. Like

* check it swaps when equal
* check if doesn't swap when unequal